tmt/steps/execute/__init__.py

```diff
@@ -157,6 +157,21 @@ def log(self, test, filename=None, full=False, create=False):
         path = os.path.join(directory, filename)
         return path if full else os.path.relpath(path, self.step.workdir)
 
+    def prepare_tests(self):
+        """
+        Prepare discovered tests for testing
+
+        Check which tests have been discovered, for each test prepare
+        aggregated metadata in a 'test.yaml' file under the test log
+        directory and finally return a list of all discovered tests.
+        """
+        tests = self.step.plan.discover.tests()
+        for test in tests:
+            self.write(
+                self.log(test, filename='test.yaml', full=True, create=True),
```

Comment:
I guess it is too late to change that but I find name of this method too confusing. log as in 'do log' instead of real log as in  'get log path'

Reply:
Yeah, see your point. Let's give it a better name (in a separate pull request).

Reply:
And here's the promised pull request for the name change: #409